integration-tests/src/test/java/oracle/kubernetes/operator/MiiBaseTest.java

```diff
@@ -76,15 +77,109 @@ public Domain createMiiDomainWithConfigMap(String domainUidPrefix,
     return domain;
   }
 
+  /**
+   * Modify the domain yaml to change domain-level restart version.
+   * @param domainNS the domain namespace
+   * @param domainUid the domain UID
+   * @param versionNo version number of domain
+   *
+   * @throws Exception if patching domain fails
+   */
+  protected void createDomainImage(Map<String, Object> domainMap, String imageName,
+                                   String modelFile, String modelPropFile) throws Exception {
+    String domainBaseImageName = (String) domainMap.get("domainHomeImageBase");
+    String domainUid = (String) domainMap.get("domainUID");
+    String domainName = (String) domainMap.get("domainName");
+    LoggerHelper.getLocal().log(Level.INFO, "imageName: " + imageName);
+    LoggerHelper.getLocal().log(Level.INFO, "domainBaseImageName: " + domainBaseImageName);
+    LoggerHelper.getLocal().log(Level.INFO, "domainUid: " + domainUid);
+    LoggerHelper.getLocal().log(Level.INFO, "domainName: " + domainName);
+
+    // Get the map of any additional environment vars, or null
+    Map<String, String> additionalEnvMap = (Map<String, String>) domainMap.get("additionalEnvMap");
+    String resultsDir = (String) domainMap.get("resultDir");
+    StringBuffer createDomainImageScriptCmd = new StringBuffer("export WDT_VERSION=");
+
+    createDomainImageScriptCmd.append(BaseTest.WDT_VERSION).append(" && ")
+      .append(getUserProjectsDir())
+      .append("/weblogic-domains/")
+      .append(domainName)
+      .append("/miiWorkDir/")
+      .append("imagetool/bin/imagetool.sh update")
+      .append(" --tag ")
+      .append(imageName)
+      .append(" --fromImage ")
+      .append(domainBaseImageName)
+      .append(" --wdtModel ")
+      .append(resultsDir)
+      .append("/samples/model-in-image/")
+      .append(modelFile)
+      .append(" --wdtVariables ")
+      .append(resultsDir)
+      .append("/samples/model-in-image/")
+      .append(modelPropFile)
+      .append(" --wdtArchive ")
+      .append(getUserProjectsDir())
+      .append("/weblogic-domains/")
+      .append(domainName)
+      .append("/miiWorkDir/models/archive.zip")
+      .append(" --wdtModelOnly ")
+      .append(" --wdtDomainType ")
+      .append(WdtDomainType.WLS.geWdtDomainType());
+
+    // patching the domain
+    LoggerHelper.getLocal().log(Level.INFO, "Command to create domain image: " + createDomainImageScriptCmd);
+    ExecResult result = ExecCommand.exec(createDomainImageScriptCmd.toString(), true, additionalEnvMap);
+    if (result.exitValue() != 0) {
+      throw new RuntimeException(
```

Comment:
this is good - we are checking the outcome, and printing out a nice descriptive message, but i think an assertion that it does not throw an exception would be better style.

Reply:
yea. I have changed it to:

catch (Exception ex) {
      StringBuffer errorMsg = new StringBuffer("FAILURE: command: ");
      errorMsg
          .append(createDomainImageScriptCmd)
          .append(" failed, returned ")
          .append(result.stdout())
          .append("\n")
          .append(result.stderr());

      Assertions.fail(errorMsg + "\n " + ex.getMessage());
    }